File: src/server/MovieTheater.Business/ViewModels/Ticket/TicketViewModel.cs
using MovieTheater.Business.ViewModels.Seat;
using MovieTheater.Business.ViewModels.Showtime;
using MovieTheater.Business.ViewModels.Movie;
using MovieTheater.Business.ViewModels.Promotion;
using MovieTheater.Models.Common;
using MovieTheater.Business.ViewModels.Room;

namespace MovieTheater.Business.ViewModels.Ticket;

public class TicketViewModel : MasterBaseViewModel
{
    public decimal Price { get; set; }
    public DateTime BookingDate { get; set; }
    public required TicketStatus Status { get; set; }
    public SeatViewModel? Seat { get; set; }
    public required string RoomName { get; set; }
    public required string MovieName { get; set; }
    public Guid InvoiceId { get; set; }
    public Guid? PromotionId { get; set; }
    public PromotionViewModel? Promotion { get; set; }
    public required DateOnly ShowDate { get; set; }
    public required TimeSpan StartTime { get; set; }
}

File: src/client/src/models/ticket/ticketview.model.ts
import { MasterBaseModel } from "../../core/models/master-base.model";
import { SeatViewModel } from "../seat/seat.model";

export class TicketViewModel extends MasterBaseModel {
    public price!: number;
    public bookingDate!: Date;
    public status!: string;
    public seat?: SeatViewModel;
    public roomName!: string;
    public movieName!: string;
    public invoiceId!: string;
    public promotionId?: string;
    // public promotion?: PromotionViewModel;
    public showDate!: Date;
    public startTime!: string;
  }
  